server/index.js

```diff
@@ -62,6 +69,27 @@ app.get('/api/me', refreshToken, (req, res, next) => {
 }
 );
 
+app.get('/api/tracksalltime', refreshToken, (req, res, next) => {
+  fetch('https://api.spotify.com/v1/me/top/tracks?time_range=long_term&limit=10', {
+    headers: {
+      Authorization: 'Bearer ' + req.user.access_token
+    }
+  })
+    .then(res => res.json())
+    .then(topTracks => {
+      const tracks = {};
+      for (let i = 0; i < topTracks.items.length; i++) {
+        tracks['key' + i] = {
+          artist: topTracks.items[i].artists[0].name,
+          track: topTracks.items[i].name,
+          image: topTracks.items[i].album.images[0].url,
+          popularity: topTracks.items[i].popularity
+        };
+      }
+      res.send(tracks);
```

Comment:
This is making your code _really heavily duplicated_

You should be able to push these into your own array.

```js
      const tracks = [];
      for (let i = 0; i < topTracks.items.length; i++) {
        tracks.push({
          artist: topTracks.items[i].artists[0].name,
          track: topTracks.items[i].name,
          image: topTracks.items[i].album.images[0].url,
          popularity: topTracks.items[i].popularity
        });
      }
```